### PR TITLE
salvage text from old GitHub wiki so we can shut it down

### PR DESCRIPTION
**Pull request by SethTisue**
I believe I've salvaged here everything salvageable from https://github.com/scala/scala/wiki, which I will shut down after this is merged.

see also https://github.com/scala/scala-lang/pull/315

### CODE REVIEW

**Comment by SethTisue** [comment]
@adriaanm I'm guessing you'll want to do a pass over the suggested-reviewer list in README.md

**Comment by som-snytt** [comment]
I hope the code style page is installed somewhere. It's much superior to the old one that is subject to so much controversy about I forget what.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,18 +38,104 @@ Please make sure the JIRA ticket's fix version corresponds to the upcoming miles
...
-   - [Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) should be applied.
+Here is some advice on how to craft a pull request with the best possible
+chance of being accepted.
 
-Please also have a look at our [Pull Request Policy](https://github.com/scala/scala/wiki/Pull-Request-Policy), as well as the [Scala Hacker Guide](http://www.scala-lang.org/contribute/hacker-guide.html) by @xeno-by.
+### Merging
+
+A pull request should merge cleanly.  (If enough time passes after
+your initial submission, we may ask you to rebase it onto the current
+mainline code.)
```
Comment (lrytz) [inline CONTRIBUTING.md]: @SethTisue there was an effort recently to keep `CONTRIBUTING.md` moderately short (https://github.com/scala/scala/pull/4298), so maybe we can drop some of the more obvious things, like the "Merging" paragraph. I think this is not a very frequent problem, and we can just ping people if necessary.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -38,18 +38,104 @@ Please make sure the JIRA ticket's fix version corresponds to the upcoming miles
...
+
+### Tests
+
+Bug fixes should include regression tests -- in the same commit as the fix.
+
+If testing isn't feasible, the commit message should explain why.
+(Consider discussing on scala-internals first.)
```
Comment (lrytz) [inline CONTRIBUTING.md]: I'll mention a few more things that could be removed IMO - I don't feel too strongly about it though, if people want to keep them in, that's fine with me.

For me, the comment in parens could go.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -38,18 +38,104 @@ Please make sure the JIRA ticket's fix version corresponds to the upcoming miles
...
+
+NOTE: it's best not to @mention in commit messages, as github pings you every time a commit with your @name on it shuffles through the system (even in other repos, on merges,...).
+
+A reviewer gives the green light by commenting "LGTM" (looks good to me).
+
+A review feedback may be addressed by pushing new commits to the request, if these commits stand on their own.
+
+Once all these conditions are met, and we agree with the change (we are available on scala-internals to discuss this beforehand, before you put in the coding work!), we will merge your changes.
+
+Please note: you are responsible for meeting these criteria  (reminding your reviewer, for example).
```
Comment (lrytz) [inline CONTRIBUTING.md]: This line could also go - we try not to let PRs hang for too long :)

File: README.md [hunk 4]
```diff
@@ -23,8 +23,26 @@ If you need some help with your PR at any time, please feel free to @-mention an
...
+* error handling, lazy vals: @hubertp (Hubert Plociniczak)
+* backend: @lrytz (Lukas Rytz), @retronym (Jason Zaugg), @dragos (Iulian Dragos)
+* repl, compiler performance: @retronym (Jason Zaugg)
+* scaladoc: @dickwall (Dick Wall), @vladureche (Vlad Ureche)
+* optimizer: @vladureche (Vlad Ureche), @lrytz (Lukas Rytz)
+* build: @jsuereth (Josh Suereth)
+* random compiler bugs: @retronym, @lrytz, @adriaanm
+* documentation: @heathermiller (Heather Miller), @dickwall (Dick Wall)
+
+P.S.: If you have some spare time to help out around here, we would be delighted to add your name to this list!
```
Comment (lrytz) [inline README.md:29]: I think we should just merge this into the above table.

Comment (lrytz) [inline README.md:29]: i think some people could be releaved from the reviewers list - unless they'd like to stay in of course! i'm thinking of @cvogt, @phaller, @hubertp, @odersky

File: README.md [hunk 5]
```diff
@@ -23,8 +23,26 @@ If you need some help with your PR at any time, please feel free to @-mention an
...
+* type checker, inference: @odersky (Martin Odersky), @adriaanm (Adriaan Moors)
+* language specification, value classes: @odersky (Martin Odersky)
+* new pattern matcher, implicit search: @adriaanm (Adriaan Moors)
+* partest: @phaller (Philipp Haller)
+* error handling, lazy vals: @hubertp (Hubert Plociniczak)
+* backend: @lrytz (Lukas Rytz), @retronym (Jason Zaugg), @dragos (Iulian Dragos)
+* repl, compiler performance: @retronym (Jason Zaugg)
+* scaladoc: @dickwall (Dick Wall), @vladureche (Vlad Ureche)
+* optimizer: @vladureche (Vlad Ureche), @lrytz (Lukas Rytz)
+* build: @jsuereth (Josh Suereth)
```
Comment (lrytz) [inline README.md]: handled by the typesafe scala team

**Comment by SethTisue** [comment]
@lrytz thanks for the pointer to #4298 which I hadn't seen before, and for the other suggestions. I've pushed a new version now. I didn't fully condense the second reviewer list into the first once yet, since I'm thinking that @adriaanm might want to cut the second one way back or just get rid of it.

**Comment by SethTisue** [comment]
ping @adriaanm. (this is neither terribly important nor urgent, but it would be nice to get it over with)

**Comment by adriaanm** [comment]
LGTM

**Comment by SethTisue** [comment]
I added one more commit that merges the second reviewer list into the first. It's possible that part could still use further tweaking, but I'm going go ahead and merge anyway — we can always continue to make edits.

**Pull request merged**